src/coreclr/src/tools/Common/JitInterface/ThunkGenerator/Program.cs

```diff
@@ -439,6 +448,271 @@ class JitInterfaceWrapper
             tw.WriteLine("};");
         }
 
+        static void WriteAPI_Names(TextWriter tw, IEnumerable<FunctionDecl> functionData)
+        {
+            tw.Write(@"// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// DO NOT EDIT THIS FILE! It IS AUTOGENERATED
+");
+
+            foreach (FunctionDecl decl in functionData)
+            {
+                tw.WriteLine($"DEF_CLR_API({decl.FunctionName})");
+            }
+
+            tw.Write(@"
+#undef DEF_CLR_API
+");
+        }
+
+        static void API_Wrapper(TextWriter tw, IEnumerable<FunctionDecl> functionData)
+        {
+
+            tw.Write(@"// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// DO NOT EDIT THIS FILE! It IS AUTOGENERATED
+
+#define API_ENTER(name) wrapComp->CLR_API_Enter(API_##name);
+#define API_LEAVE(name) wrapComp->CLR_API_Leave(API_##name);
+
+/**********************************************************************************/
+// clang-format off
+/**********************************************************************************/
+");
+
+            foreach (FunctionDecl decl in functionData)
+            {
+                tw.Write($"\n{decl.ReturnType.NativeTypeName} WrapICorJitInfo::{ decl.FunctionName}(");
+                bool isFirst = true;
+                foreach (Parameter param in decl.Parameters)
```

Comment:
nit? Looks like you could save lots of duplicated code by extracting a function to output function signatures, and function argument call lists